adrs/3122196229/README.md

```diff
@@ -0,0 +1,55 @@
+# NodeJS files and directories names conversion
+
+* **State:** Draft
+* **Created:** 2020-11-10
+* **Tags:** nodejs
+
+## Context
+
+Across the ecosystem, we can find all sort of files and directories naming
+conversions in NodeJS projects.
+
+That has caused misalignment, shifting the focus away from more important
+matters, also have created conventions that introduce problems such:
+
+* Disagreement around English language, making it even more difficult for the
+  non-English speaker to figure out the right decision.
+* Case sensitive vs. case insensitive file systems, causing issues in different
+  environments such as your local computer vs. CI/CD computers.
+* Format mismatching around separating English words.
+
+The `camelCase`, and/or `PascalCase` for naming convention, although this seems
+a good solution there are some caveats.
+
+What happen for the following cases `httpClient` vs. `HTTPClient` vs.
+`HttpClient`?
+
+* Which one is the more appropriate choice here without getting into
+  disagreement and use the phrase "it depends"?
+
+In some cases, the file `PascalCase` is used if the file is a class or
+a React component, otherwise follow `camelCase` is used.
+
+* What happen when the file contains more than a class or a React component? How
+  do we take the appropriate decision about it without getting into
+  disagreement and use the phrase "it depends"?
+
+Also, does that means that if we refactor the code, and we move the React
+component out of a file, that file name must change?
+
+That would introduce more potential refactoring like renaming file paths,
+adding more unnecessary line of code changes.
+
+As well as issues in `CI` environments, where it may be the case that `git`
```

Comment:
This is more an issue of how a developers computer is configured and we shouldn't solve file system configuration issues with naming convention.

We **should** instead make sure engineers use a case sensitive filesystem in OS X so that our dev environment more accurately reflects CI.

Reply:
That is the world where we live,

Instead of trying to fix every developer's computer out there, and infringe on their computers (especially those who contribute to open source projects for free) this intent to mitigate the issue.